### PR TITLE
Fix deleting not owned ingress

### PR DESCRIPTION
**Pull request by adamdyszy**

### CODE REVIEW

File: controllers/ibmlicensing_controller.go [hunk 1]
```diff
@@ -728,37 +728,39 @@ func (r *IBMLicensingReconciler) reconcileRouteWithTLS(instance *operatorv1alpha
...
-		reconcileResult, err := r.reconcileNamespacedResourceWhichShouldNotExist(instance, expectedIngress, foundIngress)
+		// here we want to ensure ingress doesnt exist but only if our owner reference
+		namespacedName := types.NamespacedName{Name: expectedIngress.GetName(), Namespace: expectedIngress.GetNamespace()}
+		err := r.Client.Get(context.TODO(), namespacedName, &foundIngress)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// expected behavior
+				return reconcile.Result{}, nil
+			}
+			reqLogger.Error(err, "Failed to get ingress but its ok to continue as it should not exist.", "Name", expectedIngress.GetName(),
```
Comment (jakubsolecki) [inline controllers/ibmlicensing_controller.go:754]: If it's ok not to get Ingress instance then wy do we log an error? Should we even log anything?

File: controllers/ibmlicensing_controller.go [hunk 2]
```diff
@@ -728,37 +728,39 @@ func (r *IBMLicensingReconciler) reconcileRouteWithTLS(instance *operatorv1alpha
...
+				"Namespace", expectedIngress.GetNamespace())
+			return reconcile.Result{}, nil
+		}
+		// ingress that should be deleted exists, need to check owner reference and delete if it is owned
+		if !metav1.IsControlledBy(&foundIngress, instance) {
+			reqLogger.Info("Not deleting ingress with different owner reference.")
```
Comment (jakubsolecki) [inline controllers/ibmlicensing_controller.go:760]: Same as previous. What's the point of logging this information?

Comment (adamdyszy) [reply controllers/ibmlicensing_controller.go:760]: you might forget you created ingress by urself and want to know why it is not deleting it

**Comment by Jan-Olszowka** [comment]
changes conflict with existing codebase - some pointers had been replaced with value semantics - need refactor, issue regarding this PR: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/57824

**Pull request closed without merging**